aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp

```diff
@@ -263,12 +294,86 @@ class QConvPackWeightInt8 final : public c10::OperatorKernel {
         "Didn't find engine for operation quantized::conv_prepack ",
         toString(ctx.qEngine()));
   }
+
+ private:
+#ifdef USE_FBGEMM
+  static fbgemm::conv_param_t<kSpatialDim> MakeFbgemmConvParam(
+      int C,
+      int M,
+      int group,
+      const std::vector<int>& kernel,
+      const torch::List<int64_t>& stride,
+      const torch::List<int64_t>& padding,
+      const torch::List<int64_t>& dilation);
+#endif // USE_FBGEMM
 };
 
-static auto registry = c10::RegisterOperators().op(
-    "quantized::conv_prepack",
-    c10::RegisterOperators::options().kernel<QConvPackWeightInt8>(
-        TensorTypeId::QuantizedCPUTensorId));
+#ifdef USE_FBGEMM
+
+template <>
+fbgemm::conv_param_t<2> QConvPackWeightInt8<2>::MakeFbgemmConvParam(
+    int C,
+    int M,
+    int groups,
+    const std::vector<int>& kernel,
+    const torch::List<int64_t>& stride,
+    const torch::List<int64_t>& padding,
+    const torch::List<int64_t>& dilation) {
+  return fbgemm::conv_param_t<2>(
+      1, // Mini-Batch
+      C,
+      M,
+      {28, 28}, // Image height and width
+      groups,
+      {kernel[0], kernel[1]},
+      {static_cast<int>(stride[0]), static_cast<int>(stride[1])},
+      {static_cast<int>(padding[0]),
+       static_cast<int>(padding[1]),
+       static_cast<int>(padding[0]),
+       static_cast<int>(padding[1])},
+      {static_cast<int>(dilation[0]), static_cast<int>(dilation[1])});
+}
+
+template <>
+fbgemm::conv_param_t<3> QConvPackWeightInt8<3>::MakeFbgemmConvParam(
+    int C,
+    int M,
+    int groups,
+    const std::vector<int>& kernel,
+    const torch::List<int64_t>& stride,
+    const torch::List<int64_t>& padding,
+    const torch::List<int64_t>& dilation) {
+  return fbgemm::conv_param_t<3>(
+      1, // Mini-Batch
+      C,
+      M,
+      {28, 28, 28}, // Image height and width
+      groups,
+      {kernel[0], kernel[1], kernel[2]},
+      {static_cast<int>(stride[0]),
+       static_cast<int>(stride[1]),
+       static_cast<int>(stride[2])},
+      {static_cast<int>(padding[0]),
+       static_cast<int>(padding[1]),
+       static_cast<int>(padding[2]),
+       static_cast<int>(padding[0]),
+       static_cast<int>(padding[1]),
+       static_cast<int>(padding[2])},
+      {static_cast<int>(dilation[0]),
+       static_cast<int>(dilation[1]),
+       static_cast<int>(dilation[2])});
+}
+
+#endif // USE_FBGEMM
+
+static auto registry =
+    c10::RegisterOperators()
+        .op("quantized::conv_prepack",
```

Comment:
Hmm, I don't really like the naming inconsistency here (`conv_prepack` vs `conv3d_prepack`, i'd prefer `conv2d_prepack`). It would be BC breaking but quantization is experimental so it's probably okay

Reply:
The problem is conv_prepack op is already there and if we change the api name, there may be some backward compatibility issues.